src/impl/vamp/random/xorshift.hh

```diff
@@ -5,43 +5,40 @@ extern "C"
 #include <simdxorshift128plus.h>
 }
 
-#include <array>
 #include <vamp/vector.hh>
 #include <vamp/vector/interface.hh>
+#include <vamp/random/rng.hh>
 
 namespace vamp::rng
 {
     template <std::size_t dim>
-    struct XORShift
+    struct XORShift : public RNG<dim>
     {
-        explicit XORShift(std::size_t skip_iterations = 0)
-          : XORShift(2, 3UL, 0.F, 1.F, skip_iterations) noexcept
-        {
-        }
-
         explicit XORShift(
             uint64_t key1 = 2UL,
             uint64_t key2 = 3UL,
             float min_val = 0.0,
-            float max_val = 1.0,
-            std::size_t skip_iterations = 0) noexcept
-          : min_val{min_val}, max_val{max_val}
+            float max_val = 1.0) noexcept
+          : key1_init(key1), key2_init(key2), min_val(min_val), max_val(max_val)
         {
             avx_xorshift128plus_init(key1, key2, &key);
-
-            for (auto i = 0U; i < skip_iterations; ++i)
-            {
-                next();
-            }
         }
 
-        avx_xorshift128plus_key_t key{};
+        uint64_t key1_init;
+        uint64_t key2_init;
         float min_val;
         float max_val;
+
+        avx_xorshift128plus_key_t key{};
         using IntVector = Vector<SIMDVector<__m256i>, 1, dim>;
```

Comment:
We should probably use a specialization of the `IntVector` template defined in `vector.hh` instead?